pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -21,6 +21,73 @@
 
 #define SO_SUFFIX_SIZE 3
 
+// Checkout https://datatracker.ietf.org/doc/html/rfc7540 under "HTTP/2 Connection Preface" section
+#define HTTP2_MARKER_SIZE 24
+// Checkout https://datatracker.ietf.org/doc/html/rfc7540 under "Frame Format" section
+#define HTTP2_FRAME_HEADER_SIZE 9
+// A limit of max frames we will upload from a single connection to the user mode.
+#define HTTP2_MAX_FRAMES 40
+
+
+// All types of http2 frames exist in the protocol.
+// Checkout https://datatracker.ietf.org/doc/html/rfc7540 under "Frame Type Registry" section.
+enum frame_type_t {
+    kDataFrame          = 0,
+    kHeadersFrame       = 1,
+    kPriorityFrame      = 2,
+    kRSTStreamFrame     = 3,
+    kSettingsFrame      = 4,
+    kPushPromiseFrame   = 5,
+    kPingFrame          = 6,
+    kGoAwayFrame        = 7,
+    kWindowUpdateFrame  = 8,
+    kContinuationFrame  = 9,
+};
+
+struct http2_frame {
+    uint32_t length;
+    enum frame_type_t type;
+    uint8_t flags;
+    uint32_t stream_id;
+};
+
+static __always_inline uint32_t as_uint32_t(unsigned char input) {
+    return (uint32_t)input;
+}
+
+static __always_inline bool is_empty_buffer(const char *buf, size_t buf_size) {
+    for (size_t i = 0; i < buf_size; i++) {
+        if (buf[i] != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+static __always_inline bool read_http2_frame_header(const char *buf, size_t buf_size, struct http2_frame *out) {
+    if (buf == NULL) {
+        return false;
+    }
+
+    if (buf_size < HTTP2_FRAME_HEADER_SIZE) {
+        return false;
+    }
+
+    if (is_empty_buffer((const char *)buf, HTTP2_FRAME_HEADER_SIZE)) {
```

Comment:
The cast is redundant here

```suggestion
    if (is_empty_buffer(buf, HTTP2_FRAME_HEADER_SIZE)) {
```